### PR TITLE
Added Content Status and Store View filters

### LINKED ISSUE #1463: Introduce Store view filter
Story #1291 

Add `Store view` filter to the media gallery to allow the user to filter assets by the store views where the content associate with it is used in.

The frontend input should be the same as the one used in `Content` -> `Blocks`, click in `Filters` and there is a `Store View` filter there.

In the case of entities like `Products` that are associate with a website, the filter should display the assets that are used in products that are associate with websites that have the selected store view as a child.

**Technical details**
https://github.com/magento/adobe-stock-integration/wiki/MediaGalleryUi-DataProvider

### LINKED ISSUE #1464: Introduce Enabled filter
Story #1291 

Add a filter to allow the user to search by the status of the entity where assets are used in, for example, user filters by enabled, the filter should search across all entities (products, blocks, pages, and categories) and return assets that are used in entities that are enabled.

The frontend input should be the same used across other grids:

Screenshot 2020-06-22 at 14 39 31

**Technical details**
https://github.com/magento/adobe-stock-integration/wiki/MediaGalleryUi-DataProvider

### PR DESCRIPTION
**Pull request by gabrieldagama**
### Description (*)
Added Content Status and Store View filters.

1. Depends on https://github.com/magento/magento2/pull/29058

### Related Pull Requests
https://github.com/magento/magento2/pull/29058

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1463 
2. Fixes magento/adobe-stock-integration#1464

### Manual testing scenarios (*)

1. Assign an asset to a content (product, category, page or block)
2. Go to `Content` -> `Media Gallery` and filter by `Content Status` or `Store View`
3. The filter output should be assets that are used on content and the content matches the filter condition.

### CODE REVIEW

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

File: MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/ContentField.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor\CustomFilterInterface;
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\MediaContentApi\Api\GetAssetIdsByContentFieldInterface;
+
+/**
+ * Class responsible to filter a content field
+ */
+class ContentField implements CustomFilterInterface
+{
+    private const TABLE_ALIAS = 'main_table';
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/ContentField.php]: A bit of inconsistency: there is a constant for the table name but no constant for `id` field name. I'd have both on none

File: MediaGalleryUi/Test/Mftf/ActionGroup/AdminMediaGalleryApplySelectFilterActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminMediaGalleryApplySelectFilterActionGroup">
+        <annotations>
+            <description>Applies select filter to the media gallery grid</description>
+        </annotations>
+        <arguments>
+            <argument name="selectLabel" type="string"/>
+            <argument name="selectOption" type="string"/>
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Mftf/ActionGroup/AdminMediaGalleryApplySelectFilterActionGroup.xml]: `filterLabel` and `optionLabel` argument names would be more descriptive for a developer

File: MediaGalleryUi/Test/Mftf/Section/AdminEnhancedMediaGalleryFiltersSection.xml [hunk 3]
```diff
@@ -16,7 +16,8 @@
         <element name="usedInSelectDropdown" type="text" selector="//label[@class='admin__form-field-label']/span[text()='Show Images Used In']/parent::*/parent::div/div//div[@class='admin__action-multiselect-text' and text()='Select...']"/>
         <element name="usedInEntityType" type="text" selector="//label[@class='admin__action-multiselect-label']/span[text()='{{entityType}}']" parameterized="true"/>
         <element name="usedInDoneButton" type="button" selector="//div[@class='admin__action-multiselect-actions-wrap']/button/span[text()='Done']"/>
-         <element name="duplicatedFilterCheckbox" type="button" selector="//input[@name='duplicated']"/>
-
+        <element name="selectFilterInput" type="button" selector="//label[@class='admin__form-field-label']/span[text()='{{filterLabel}}']/parent::*/parent::div/div[@class='admin__form-field-control']/select" parameterized="true"/>
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Mftf/Section/AdminEnhancedMediaGalleryFiltersSection.xml]: Why do these selectors have `Input` in  the name?

File: MediaGalleryUi/Test/Mftf/ActionGroup/AdminOpenMediaGalleryForExistingPageNoEditorActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminOpenMediaGalleryForExistingPageNoEditorActionGroup">
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Mftf/ActionGroup/AdminOpenMediaGalleryForExistingPageNoEditorActionGroup.xml]: Can you please replace existing usages of `AdminOpenMediaGalleryForPageNoEditorActionGroup` with opening a page and using this action group.

Also I'd correct a name as I believe this action group should work not only for the existing page
```suggestion
    <actionGroup name="AdminOpenMediaGalleryFromPageNoEditorActionGroup">
```

File: MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryStoreViewContentFilterTest.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminMediaGalleryStoreViewContentFilterTest">
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryStoreViewContentFilterTest.xml:10]: I'd also add a store view filter test for the category to cover the specific implementation

File: modman [hunk 6]
```diff
@@ -19,6 +19,7 @@
 @import MediaContentSynchronizationApi
 @import MediaContentSynchronizationCatalog
 @import MediaContentSynchronizationCms
+@import MediaGalleryContentUi
```
Comment (sivaschenko) [inline modman]: Please remove the obsolete import

File: MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryDisabledContentFilterTest.xml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminMediaGalleryDisabledContentFilterTest">
+        <annotations>
+            <features value="MediaGallery"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/1464"/>
+            <title value="User filter asset by disabled content"/>
+            <stories value="Story 57: User filters images by the area they used in"/>
+            <testCaseId value="to-be-added"/>
```
Comment (chalov-anton) [inline MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryDisabledContentFilterTest.xml]:
```suggestion
            <testCaseId value="https://studio.cucumber.io/projects/131313/test-plan/folders/1320712/scenarios/4970565"/>
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @gabrieldagama , please see my comments

**Comment by gabrieldagama** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by chalov-anton** [comment]
:heavy_check_mark: Great job

The _Store view_ filter handles multiple Sites and Views perfectly 
store_view

The _Content status_ filter works for Enabled and Disabled Pages, Blocks, Products, Categories

The filters work together with other already existing filters

**Pull request merged**